gen3-cli/cot/backend/query/__init__.py

```diff
@@ -0,0 +1,98 @@
+import os
+import json
+import collections
+import jmespath
+from tabulate import tabulate
+from jmespath.exceptions import JMESPathError
+from cot.backend.create import blueprint
+from cot.backend.common import context
+from cot.backend.common import exceptions
+
+
+def run(
+    blueprint_generation_input_source=None,
+    blueprint_generation_provider=None,
+    blueprint_generation_framework=None,
+    blueprint_generation_scenarios=None,
+    blueprint_refresh=None,
+    list_tiers=None,
+    list_components=None,
+    query=None,
+    cwd=None,
+    _is_cli=False
+):
+    ctx = context.Context(cwd)
+    output_dir = os.path.join(ctx.cache, 'query', os.path.relpath(ctx.directory, ctx.root))
+    blueprint_filename = os.path.join(output_dir, 'blueprint.json')
+    # Due to the fact that it's hard to dermine refresh condition
```

Comment:
may be blueprint itself has some version or creation date information? otherwise the only way I see it is to get hashes of the files, but it's quite heavy operation to be used this way

Reply:
It doesn't really matter because the only way we can compare blueprints is by generating a new one. It's possible to use git commit hashes for that, but right now I'm unsure about how exactly git will be integrated into CMDB.